### PR TITLE
Complete task 10 CORE: GET /api/users

### PR DESCRIPTION
**Pull request by Dimeben**

### CODE REVIEW

File: __tests__/users.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+beforeEach(() => {
+  return seed({ topicData, userData, articleData, commentData });
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("/api/users", () => {
+  test("200 - GET - will return an article object with the given article_id, with the properties of author, title, article_id, body, topic, created_at, votes and article_img_url", () => {
```
Comment (nc-verity-gregory) [inline __tests__/users.test.js:21]: Is this what you're testing for? Sounds like you want users not an article object. Don't copy and paste tests, you need to be able to know how to write them.

File: __tests__/users.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        expect(users).toHaveLength(4);
+        expect(Array.isArray(users)).toBe(true);
+        users.forEach((user) => {
+          expect(user).toHaveProperty("username", expect.any(String));
+          expect(user).toHaveProperty("name", expect.any(String));
+          expect(user).toHaveProperty("avatar_url", expect.any(String));
+        });
+      });
+  });
+  test("404 - GET - sends an appropriate status and error message when an invalid URL is passed", () => {
```
Comment (nc-verity-gregory) [inline __tests__/users.test.js:36]: You have already tested for this, this end point cannot 404. All invalid endpoints will 404 and this is not unique to this end point.

**Review by nc-verity-gregory — COMMENTED** [review]
You can tidy up the error handler here, you've duplicated it. Other than that, good job.

File: __tests__/users.test.js (hunk 2 above)
Comment (Dimeben) [reply __tests__/users.test.js:36]: Screenshot 2024-08-27 at 14 22 17

**Pull request merged**